asserts/database.go

```diff
@@ -176,3 +192,94 @@ func (db *Database) Check(assert Assertion) error {
 	}
 	return fmt.Errorf("failed signature verification: %v", lastErr)
 }
+
+func (db *Database) readAssertion(assertType AssertionType, primaryPath string) (Assertion, error) {
+	encoded, err := db.readEntry(assertionsRoot, string(assertType), primaryPath)
+	if os.IsNotExist(err) {
+		return nil, ErrNotFound
+	}
+	if err != nil {
+		return nil, fmt.Errorf("broken assertion storage, failed to read assertion: %v", err)
+	}
+	assert, err := Decode(encoded)
+	if err != nil {
+		return nil, fmt.Errorf("broken assertion storage, failed to decode assertion: %v", err)
+	}
+	return assert, nil
+}
+
+func (db *Database) readCurrentRevision(assertType AssertionType, primaryPath string) (int, error) {
+	assert, err := db.readAssertion(assertType, primaryPath)
+	if err == ErrNotFound {
+		return -1, nil
```

Comment:
Okay, so let's get rid of readCurrentRevision altogether, because it's really not adding much on top of readAssertion. More importantly, though: can we please establish this rule for future work: 

_Never return invalid data if err == nil._